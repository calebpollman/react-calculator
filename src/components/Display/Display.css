.display-container {
  align-items: center;
  background: var(--display-background-color);
  display: flex;
  height: var(--display-height);
  padding: 0 4%;
  width: var(--display-width);
}

.display-value {
  color: var(--display-text-color);
  font-size: var(--display-text-size);
  font-family: var(--main-font);
  font-weight: var(--display-text-weight);
  margin-left: auto;
  overflow: hidden;
}
