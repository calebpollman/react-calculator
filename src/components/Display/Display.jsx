import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ({ displayValue }) => (
  <div className="display-container">
    <p className="display-value">
      {displayValue}
    </p>
  </div>
);

Display.propTypes = { displayValue: PropTypes.string.isRequired };

export default Display;
