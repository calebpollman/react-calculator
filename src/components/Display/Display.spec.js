import { shallow } from 'enzyme';
import Display from './Display';

const MOCK_DISPLAY_VALUE = 'a display value';

describe('Display', () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Display displayValue={MOCK_DISPLAY_VALUE} />)));

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('renders the value of displayValue', () => {
    expect(wrapper.text()).toEqual(MOCK_DISPLAY_VALUE);
  });
});
