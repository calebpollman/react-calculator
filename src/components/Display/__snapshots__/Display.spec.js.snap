// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render correctly 1`] = `
<div
  className="display-container"
>
  <p
    className="display-value"
  >
    a display value
  </p>
</div>
`;
